Documentation/Relationship/Relationship.md

```diff
@@ -0,0 +1,333 @@
+  "lastname": "Raser",
+  "email": "rudi@raser.de",
+  "_links": {
+    "self": "https://example.api/user/532",
+    "post": "https://example.api/post"
+  }
+}
+```
+
+One of the core concepts of REST (with HATEOAS) is that a client does not need to know anything about the path structure.
+Instead, just like a web browser, a client follows hyperlinks returned from the web service.  
+This allows you to change the path structure of your REST service - keeping the same named links at the same places -
+without breaking compatibility for your clients.
+
+To support such scenarios there must be some sort of way to manually overwrite such relationship information.
+
+## 2. Manually defining "structural" relationship information
+
+This chapter imagines ways how a user could manually override or provide additional relationship information.
+
+### 2.1. Overriding the relationship name
+
+The above described inferring approach will use the string description of the `PathComponent` as the relationship name.
+This may not fit everybody's needs.
+
+Similar to how you can manually define a `@Parameter` name, you can also override a relationship name.
+This proposal suggest a sort of `PathComponentModifier` (similar to `ComponentModifiers`).
+
+Such a definition could then look like the following:
+```swift
+var content: some Component {
+  Group("test".relationshipName("new-name")) {
+    Handler()
+  }
+}
+```
+
+### 2.2. Adding new relationships
+
+In order to support rearranging components, a user can also define its own relationships.
+We introduce a `.relationship` modifier for `PathComponent`s, to which one would pass
+a relationship name and `PathComponent`s describing the destination.  
+The destination can of course also include `@PathParameters`.
+
+Any manual definition of a relationship MUST always override a inferred relationship if the name collides.
+
+Looking at the example below, a REST exporter would add the link `greeter` pointing to `/greeting/{userid}` on a 
+response returned from `/user/:userId`. As the value of the `userId` `PathParameter` is known, it is replaced with
+the actual value. `@PathParameter`s for which no predefined value can be inferred should be formatted according to
+[RFC 6570](https://tools.ietf.org/html/rfc6570) (as well as potential query parameters of the destination). 
+
+```swift
+var content: some Component {
+  Group("user", $userId.relationship("greeter", to: "greeting", $userId)) {
+    Handler()
+  }
+  Group("greeting", $userId) {
+    Greeter()
+  }
+}
+```
+
+### 2.3. Excluding relationships
+
+It may also be a sensible decision for a user to hide a inferred relationship. To support that
+a modifier can be introduce.
+
+Even if a user supplies a `.relationshipName` modifier, the relationship will still be hidden when `.hideRelationship` is used.
+
+Considering the example below, a request to `/` will not include a relationship pointing to `/test`.
+
+```swift
+var content: some Component {
```

Comment:
I think this option may be specifically tailored towards REST.

When having the web service defined like in that example, a request to `/` won't include a link to `/test` in the `_links` section, but a request pointing to `/test` will nonetheless be successful.
So graphql would also just server a query to `query { test { ... } }`.

The more I think about this, the harder it is to find a real use case for this 🤔

Reply:
Maybe we should add a `Maybe` section (like the generic manifesto from Swift: [here](https://github.com/apple/swift/blob/main/docs/GenericsManifesto.md#maybe)).

So that we still document the idea but don't commit to it necessarily ;)

Reply:
I think that is a good idea. Would suggest to move both `2.2` and `2.3` to the Maybe section.